src/navigationGuards.ts

```diff
@@ -253,6 +253,13 @@ export function extractComponentsGuards(
           )
           let promise = rawComponent
           rawComponent = () => promise
+        } else if ((rawComponent as any)['__asyncLoader']) {
+          let asyncComponent = rawComponent
+          rawComponent = () => Promise.resolve(asyncComponent)
+          warn(
+            `Async component in record with path "${record.path}" can not be used in routes. ` +
```

Comment:
I would prefer a warning that points what to do:

```js
`Component "${name}" in record with path "${record.path}" is defined ` +
              `using "defineAsyncComponent()". ` +
              `Write "() => import('./MyPage.vue')" instead of ` +
              `"defineAsyncComponent(() => import('./MyPage.vue'))".`
```